src/library_manager/llext_manager.c

```diff
@@ -51,6 +51,15 @@ extern struct tr_ctx lib_manager_tr;
 
 #define PAGE_SZ		CONFIG_MM_DRV_PAGE_SIZE
 
+static int llext_manager_update_flags(void __sparse_cache *vma, size_t size, uint32_t flags)
+{
+	size_t pre_pad_size = (uintptr_t)vma & (PAGE_SZ - 1);
+	void *aligned_vma = (__sparse_force uint8_t *)vma - pre_pad_size;
```

Comment:
ALIGN_DOWN

Reply:
@softwarecki you could use it, but I also need `pre_pad_size`, so I open code it.